blocks/scripts/plot.py

```diff
@@ -0,0 +1,188 @@
+
+from __future__ import division, print_function
+
+import dill
+import fnmatch
+import logging
+import readline
+import pandas
+
+from six import iteritems
+from six.moves import input
+from collections import OrderedDict
+from functools import reduce
+
+from blocks import config
+from blocks.utils import change_recursion_limit
+
+
+def match_channel_specs(experiments, channel_specs):
+    """Filter a dictionary with experiments according to channel_specs.
+
+    Parameters
+    ----------
+    experiments : OrderedDict of {str: DataFrames}
+    channel_specs : list of str
+
+    Returns
+    -------
+        Returns a single DataFrame containing the specified
+        channels as columns.
+
+    """
+    # We iterate over all channel_specs and match each spec to the
+    # channels of all experiments.
+    df = pandas.DataFrame()
+    for spec in channel_specs:
+        if ":" in spec:
+            exp_spec, channel_spec = spec.split(":")
+        else:
+            exp_spec, channel_spec = None, spec
+
+        for i, exp in enumerate(experiments.values()):
+            for channel in fnmatch.filter(exp.columns, channel_spec):
+                if exp_spec and exp_spec != i:
+                    # We are looking for a specific experiment..
+                    #  ... and it's not this one.
+                    continue
+
+                column_name = "{}:{}".format(i, channel)
+                df[column_name] = exp[channel]
+
+    return df
+
+
+def plot_dataframe(dataframe):
+    import pylab
+
+    t = dataframe.index
+    print("Plotting {} channels:".format(len(dataframe.columns)))
+    for cname, series in iteritems(dataframe):
+        print("    {}".format(cname))
+        pylab.plot(t, series, label=cname)
+    pylab.legend()
+    pylab.show(block=True)
+
+
+PROMPT_HEADER = """
+
+Type a comma separated list of channels to plot or [q]uit.
+
+Channels may be prefixed by <number>: to refer to a specific experiment \
+and may contain '*' or '?' characters to match multiple channels at once.
+
+"""
+
+IPYTHON_HEADER = """
+
+Complete DataFrames for all experiments can be found in the \
+'experiments[fname]' dictionary. The DataFrame containing only \
+the selected channels can be access as 'matched'
+
+"""
+
+
+def plot(exp_fnames, channel_specs, list_only=False, ipython=False):
+    # Load and convert experiments into DataFrames...
+    experiments = OrderedDict()
+    for fname in exp_fnames:
+        logging.info("Loading '{0}'...".format(fname))
+        try:
+            # TODO: Load "dumped" experiments
+            with change_recursion_limit(config.recursion_limit):
+                main_loop = dill.load(open(fname, "rb"))
+        except EnvironmentError:
+            logging.error("Could not open '{}'".format(fname))
+
+        rows_to_keep = [0] + main_loop.log._status._epoch_ends
```

Comment:
Here many people would also need plotting all iterations and some sort of rolling average, because on certain datasets an epoch takes a day. But I do not have to implement this, somebody else (very likely me, because I need that a lot) can do it.